utilities/tools/Clear-ManagementGroupDeployment.ps1

```diff
@@ -0,0 +1,96 @@
+﻿
+<#
+.SYNOPSIS
+Bulk delete all deployments on the given management group scope
+
+.DESCRIPTION
+Bulk delete all deployments on the given management group scope
+
+.PARAMETER ManagementGroupId
+Mandatory. The Resource ID of the Management Group to remove the deployments for.
+
+.PARAMETER DeploymentStatusToExclude
+Optional. The status to exlude from removals. By default, we exclude any deployment that is in state 'running'.
+
+.EXAMPLE
+Clear-ManagementGroupDeployment -ManagementGroupId 'MyManagementGroupId'
+
+Bulk remove all 'non-running' deployments from the Management Group with ID 'MyManagementGroupId'
+#>
+function Clear-ManagementGroupDeployment {
+
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $ManagementGroupId,
+
+        [Parameter(Mandatory = $false)]
+        [string] $DeploymentStatusToExclude = 'running'
+    )
```

Comment:
I'd suggest to add a boolean parameter to allow skipping the removal of failed deployments, useful for troubleshooting. E.g.
``` 
        [Parameter(Mandatory = $false)]
        [bool] $SkipFailedDeployments = $true
```

Reply:
Now that it is an array, sure. I'm anyway still in doubt if we shouldn't keep failed deployments by default. I'd go for it but it's also true we could eventually reach the same 800 limit with failed deployments only. What do you think?

Reply:
Folks will eventually run of ouf deployments and need to remove failed ones too. However, I'm fine with keeping them by default until this limit is actually reached. I guess the pipeline we want as part of CARML could have a flag that deletes them too if a user reaaaally wants to.

Reply:
done